### PR TITLE
fix: fix search bar

### LINKED ISSUE #21: Search Broken
### Overview
- Search broken.
- I already updated the dependencies (we are using the `ubiquity` branch for dao.ubq.fi

### Context
```

did some research and found https://github.com/NotionX/react-notion-x/pull/505

this is a dependency of “nextjs-notion-starter-kit” which i presume was updated

our last sync was june 9 (before the fix) https://github.com/pavlovcik/dao.ubq.fi

```

### Error 
```
POST https://dao.ubq.fi/api/search-notion 405
```

### PR DESCRIPTION
**Pull request by rndquu**
Resolves https://github.com/pavlovcik/dao.ubq.fi/issues/21

### CODE REVIEW

**Review by 0x4007 — COMMENTED** [review]
What's the story to this? Is this a custom fix or did you lift it from somewhere? 

Looks very specific.

File: lib/notion.ts [hunk 1]
```diff
@@ -64,5 +64,38 @@ export async function getPage(pageId: string): Promise<ExtendedRecordMap> {
 }
 
 export async function search(params: SearchParams): Promise<SearchResults> {
-  return notion.search(params)
+  // NOTICE: Refactored the original implementation because the search
+  // feature is broken in the latest "notion-client@6.16.0" (https://github.com/NotionX/react-notion-x/pull/505)
+  // TODO: Refactor the "search()" methof to its original implementation
+  // https://github.com/transitive-bullshit/nextjs-notion-starter-kit/blob/7193796abd714356d5bb8e621b0f422e0b89ec03/lib/notion.ts#L66
```
Comment (0x4007) [inline lib/notion.ts:70]: Hard to see comment links from my phone but I'll try the deployment to test.

**Review by 0x4007 — APPROVED** [review]

**Comment by 0x4007** [comment]
Thanks for the fix!

**Comment by rndquu** [comment]
> What's the story to this? Is this a custom fix or did you lift it from somewhere?
> 
> Looks very specific.

The search bar is broken in the latest `react-notion-x` npm package but fixed in the github's master branch. So basically we should wait for the new npm package build.

This PR simply takes the [original implementation](https://github.com/Armster15/react-notion-x/blob/f7daab7ff04c634e17bd3b36aa78d4060d18c665/packages/notion-client/src/notion-api.ts#L544) of how the search works under the hood + fixes the notion search bug.

**Pull request merged**